repositories/MerchOrderRepository.ts

```diff
@@ -62,3 +63,40 @@ export class OrderItemRepository extends BaseRepository<OrderItemModel> {
     return count > 0;
   }
 }
+
+@EntityRepository(OrderPickupEventModel)
+export class OrderPickupEventRepository extends BaseRepository<OrderPickupEventModel> {
+  public async getPastPickupEvents(): Promise<OrderPickupEventModel[]> {
+    return this.getBaseFindQuery()
+      .where('"end" < :now')
+      .setParameter('now', new Date())
+      .getMany();
+  }
+
+  public async getFuturePickupEvents(): Promise<OrderPickupEventModel[]> {
+    return this.getBaseFindQuery()
+      .where('"end" >= :now')
+      .setParameter('now', new Date())
+      .getMany();
+  }
+
+  public async findByUuid(uuid: Uuid): Promise<OrderPickupEventModel> {
+    return this.getBaseFindQuery().where({ uuid }).getOne();
+  }
+
+  public async upsertPickupEvent(pickupEvent: OrderPickupEventModel,
+    changes?: Partial<OrderPickupEventModel>): Promise<OrderPickupEventModel> {
+    if (changes) pickupEvent = OrderPickupEventModel.merge(pickupEvent, changes);
+    return this.repository.save(pickupEvent);
+  }
```

Comment:
Right now we're not passing any `changes` into this, so we can do
```suggestion
  public async upsertPickupEvent(pickupEvent: OrderPickupEventModel): Promise<OrderPickupEventModel> {
    return this.repository.save(pickupEvent);
  }
```